stdlib/filename.mli

```diff
@@ -15,6 +15,9 @@
 
 (** Operations on file names. *)
 
+val null : string
```

Comment:
I would not put this as the first export in `filename.mli`.  This is not what users of the `Filename` module are looking for with highest priority. 

What about exporting and documenting it between `dirname` and `temp_file`?  `temp_file` is about generating file names for output, and `null` is also a file name for output, to some extent.

Reply:
Indeed; I amended as suggested. Thanks!

Reply:
If you are happy with the state of the PR, could you formally "accept" it? Thanks!